src/EditorFeatures/Core/Remote/SolutionChecksumUpdater.cs

```diff
@@ -142,17 +139,15 @@ private void OnWorkspaceChanged(object? sender, WorkspaceChangeEventArgs e)
 
     private async ValueTask SynchronizePrimaryWorkspaceAsync(CancellationToken cancellationToken)
```

Comment:
this is called, serially, by the ABWQ.  So we never have to be concerned about the workspace version soemhow going downwards.